enterprise/server/backends/redis_cache/redis_cache.go

```diff
@@ -157,23 +157,35 @@ func (c *Cache) Contains(ctx context.Context, d *repb.Digest) (bool, error) {
 	return found, err
 }
 
+// TODO(buildbuddy-internal#1485) - Add last access time
+// Note: Can't use rdb.ObjectIdleTime to calculate last access time, because rdb.StrLen resets the idle time to 0,
+// so this API will always incorrectly set the last access time to the current time
 func (c *Cache) Metadata(ctx context.Context, d *repb.Digest) (*interfaces.CacheMetadata, error) {
 	key, err := c.key(ctx, d)
 	if err != nil {
 		return nil, err
 	}
-	found, err := c.rdb.Expire(ctx, key, ttl).Result()
+	found, err := c.rdb.Exists(ctx, key).Result()
 	if err != nil {
 		return nil, err
 	}
-	if !found {
+	if found == 0 {
 		return nil, status.NotFoundErrorf("Digest '%s/%d' not found in cache", d.GetHash(), d.GetSizeBytes())
 	}
 	blobLen, err := c.rdb.StrLen(ctx, key).Result()
 	if err != nil {
 		return nil, err
 	}
-	return &interfaces.CacheMetadata{SizeBytes: blobLen}, nil
+
+	lastModifyTimeSec, err := c.rdb.LastSave(ctx).Result()
```

Comment:
LastSave returns the timestamp when the DB was written to disk. Our default configuration does not enable persistence so this is not going to return anything meaningful.

Reply:
Ah good catch. Removed it